File: Data Structures & Algorithms/linked-list-cycle-detection/submission-1.swift
/**
 * Definition for singly-linked list.
 * class ListNode {
 *     var val: Int
 *     var next: ListNode?
 *     init(_ val: Int) {
 *         self.val = val
 *         self.next = nil
 *     }
 * }
 */

class Solution {
    func hasCycle(_ head: ListNode?) -> Bool {
        var fast = head 
        var slow = head 

        while fast != nil && fast?.next != nil {
            slow = slow?.next // 1 
            fast = fast?.next?.next // 2
            if slow === fast {
                return true 
            }
        }
        return false 
    }
}

File: Data Structures & Algorithms/remove-linked-list-elements/submission-1.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:

        # remove all nodes == to val (target)
        # dummy node 
        # no head 
        if not head:
            return None

        dummy = ListNode(-1)
        dummy.next = head 
        current = dummy 

        while current.next:
            # if target 
            if current.next.val == val:
                current.next  = current.next.next 
            else:
                current = current.next 
        return dummy.next  

        